addons/sale/models/sale.py

```diff
@@ -745,12 +745,15 @@ def _compute_qty_delivered_updateable(self):
                  'invoice_lines.invoice_id.refund_invoice_ids.state',
                  'invoice_lines.invoice_id.refund_invoice_ids.amount_total')
     def _compute_invoice_amount(self):
+        refund_lines_product ={p.id: set() for p in self.mapped('product_id')}
         for line in self:
             # Invoice lines referenced by this line
             invoice_lines = line.invoice_lines.filtered(lambda l: l.invoice_id.state in ('open', 'paid'))
             # Refund invoices linked to invoice_lines
             refund_invoices = invoice_lines.mapped('invoice_id.refund_invoice_ids').filtered(lambda inv: inv.state in ('open', 'paid'))
-            refund_invoice_lines = refund_invoices.mapped('invoice_line_ids').filtered(lambda l: l.product_id == line.product_id)
+            refund_invoice_lines = refund_invoices.mapped('invoice_line_ids').filtered(lambda l: l.product_id == line.product_id and l.id not in refund_lines_product[line.product_id.id])
+            if refund_invoice_lines:
+                refund_lines_product[line.product_id.id] = refund_lines_product[line.product_id.id].union(set(refund_invoice_lines.ids))
```

Comment:
Could maybe be shorter:
`refund_lines_product[line.product_id.id] |= set(refund_invoice_lines.ids)`